### ISSUE TITLE
'dict' object has no attribute 'decode' when using GraphAPI.for_application

### ISSUE THREAD

**Issue by EpocDotFr** [jgorset/facepy#140]
Python 3.5.1

I'm using facepy to allow an application to access a page feed. I firstly get the OAuth access token for my application with the right App ID and App Secret (defined in my environment variables):

```python
facebook = GraphAPI.for_application(
    env('FACEBOOK_APP_ID'),
    env('FACEBOOK_APP_SECRET'),
    api_version='2.5'
)
```
At this point, if I launch my script, I get this error:

`AttributeError: 'dict' object has no attribute 'decode'` in `urllib\parse.py` line 98.

The aforementioned line: `return tuple(x.decode(encoding, errors) if x else '' for x in args)`.

It seems that this error is subsequently raised when `parse_sq` is called [here](https://github.com/jgorset/facepy/blob/master/facepy/utils.py#L63) in facepy but I cannot tell you the Facebook HTTP response content, I don't know how to get it.

Full strack trace:

```
    self.facebook = GraphAPI.for_application(env('FACEBOOK_APP_ID'), env('FACEBOOK_APP_SECRET'), api_version='2.5')
  File "C:\Python35\lib\site-packages\facepy\graph_api.py", line 48, in for_application
    access_token = get_application_access_token(id, secret_key, api_version=api_version)
  File "C:\Python35\lib\site-packages\facepy\utils.py", line 63, in get_application_access_token
    data = parse_qs(response)
  File "C:\Python35\lib\urllib\parse.py", line 576, in parse_qs
    encoding=encoding, errors=errors)
  File "C:\Python35\lib\urllib\parse.py", line 607, in parse_qsl
    qs, _coerce_result = _coerce_args(qs)
  File "C:\Python35\lib\urllib\parse.py", line 114, in _coerce_args
    return _decode_args(args) + (_encode_result,)
  File "C:\Python35\lib\urllib\parse.py", line 98, in _decode_args
    return tuple(x.decode(encoding, errors) if x else '' for x in args)
  File "C:\Python35\lib\urllib\parse.py", line 98, in <genexpr>
    return tuple(x.decode(encoding, errors) if x else '' for x in args)
AttributeError: 'dict' object has no attribute 'decode'
```

**Comment by EpocDotFr** [comment]
ATM I changed the way I use the Graph API to post new statuses on a page, and it's working well. I ended to not using `GraphAPI.for_application`.

**Comment by jgorset** [comment]
Thanks for reporting, @EpocDotFr! I think I've narrowed it down to Facebook changing the response format of this endpoint.

```http
GET /oauth/access_token?client_id=160327664029652&client_secret=102d4e42d228d59c7ae4ebd874ef7757&grant_type=client_credentials HTTP/1.1
Host: graph.facebook.com
Connection: close
User-Agent: Paw/2.2.9 (Macintosh; OS X/10.11.2) GCDHTTPRequest


HTTP/1.1 200 OK
Access-Control-Allow-Origin: *
Content-Type: text/plain; charset=UTF-8
X-FB-Trace-ID: HgLBl75uZw/
X-FB-Rev: 2146061
Pragma: no-cache
Cache-Control: private, no-cache, no-store, must-revalidate
Facebook-API-Version: v2.0
Expires: Sat, 01 Jan 2000 00:00:00 GMT
X-FB-Debug: ZTovtRGGBlRanS2YGSFT8jwsffItAFfIS7j/r6EZKWjVQ7PLJZ8466orq/oYlDCmUJceDUx0fNDzuJMx25UeCw==
Date: Tue, 26 Jan 2016 18:51:05 GMT
Connection: close
Content-Length: 56

access_token=160327664029652|3X6oBcI7twccbN7P75E-J94IYeE
```

```http
GET /v2.5/oauth/access_token?client_id=160327664029652&client_secret=102d4e42d228d59c7ae4ebd874ef7757&grant_type=client_credentials HTTP/1.1
Host: graph.facebook.com
Connection: close
User-Agent: Paw/2.2.9 (Macintosh; OS X/10.11.2) GCDHTTPRequest


HTTP/1.1 200 OK
Access-Control-Allow-Origin: *
Content-Type: application/json; charset=UTF-8
X-FB-Trace-ID: Bc+Be0PXPaG
X-FB-Rev: 2146061
Pragma: no-cache
Cache-Control: private, no-cache, no-store, must-revalidate
Facebook-API-Version: v2.5
Expires: Sat, 01 Jan 2000 00:00:00 GMT
X-FB-Debug: ncWWOXkilAxGr0cmSOAJQRlmuArF2529x3DRqFk7KqgjWYwYfVojoYfOIttM0zl6pRYntI9uaH/4x15ftEZzjQ==
Date: Tue, 26 Jan 2016 18:51:51 GMT
Connection: close
Content-Length: 84

{"access_token":"160327664029652|3X6oBcI7twccbN7P75E-J94IYeE","token_type":"bearer"}
```

This change seems to have happened in v2.3. [`facepy.utils.get_application_access_token`](https://github.com/jgorset/facepy/blob/master/facepy/utils.py#L47) should try to parse it as a query string, then, and fall back on parsing it as JSON (like [`get_extended_access_token`](https://github.com/jgorset/facepy/blob/master/facepy/utils.py#L32) already does).

**Comment by jgorset** [comment]
I fixed it and released 1.0.8.

**Comment by EpocDotFr** [comment]
Thank you!
